backend/e4h-services/ingestion-service/app/api/endpoints/file_ingestion.py

```
@@ -989,4 +990,107 @@ async def check_duplicate_tickets(
+                update_payload = create_update_payload(search_response, update_data, subtype_mapping)
+                update_response = incident_client.update_incident(update_payload)
+
+                process_update_response(update_response, df, index, update_data)
+
+            except Exception as e:
+                df.at[index, 'status'] = 'failed'
+                df.at[index, 'error'] = str(e)
+
+        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+        output_filename = f"incident_update_results_{timestamp}.xlsx"
+        output_temp_file = tempfile.NamedTemporaryFile(delete=False, suffix=".xlsx")
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)